Libtool uses links to handle locking object files with 
dependencies.  Hard links can't cross filesystems though, 
so we have to use a diff source for the link.

http://bugs.gentoo.org/40992

--- libtool-1.5.10/ltmain.in
+++ libtool-1.5.10/ltmain.in
@@ -1005,7 +1005,7 @@
     # Lock this critical section if it is needed
     # We use this script file to make the link, it avoids creating a new file
     if test "$need_locks" = yes; then
-      until $run ln "$progpath" "$lockfile" 2>/dev/null; do
+      until $run ln "$srcfile" "$lockfile" 2>/dev/null; do
 	$show "Waiting for $lockfile to be removed"
 	sleep 2
       done
